src/sagemaker/image_uris.py

```diff
@@ -22,6 +22,7 @@
 logger = logging.getLogger(__name__)
 
 ECR_URI_TEMPLATE = "{registry}.dkr.{hostname}/{repository}:{tag}"
+MONITOR_URI_TEMPLATE = "{registry}.dkr.{hostname}/{repository}"
```

Comment:
I'd prefer the code in this file be as framework/algorithm-agnostic as possible. so let's remove the tag part from the template, and then figure out how to structure the JSON instead?

this is a bit tough since I hadn't even considered that there might be URIs without tags when I started this...you could use `""` for the version? or add some logic to handle the case where `"versions"` isn't a valid key.

Reply:
I updated the model-monitor.json file as well as the ``retrieve()`` logic to let model monitor use the generic template.

We could update the "versioins" later when model-monitor images with tags are released.